_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+
+```yml
+data:
+  application.yml: |-
+    ...
+    spring:
+      security:
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: <client-secret>
+```
+
+Enable the OIDC authentication in the `jhipster-registry` service by adding the `oauth2` profile in the `kubernetes/registry-k8s/jhipster-registry.yml` file:
+
+```yml
+- name: SPRING_PROFILES_ACTIVE
+  value: prod,k8s,oauth2
+```
+
+We are going to build the `gateway` and `store` in a continuous integration workflow with CircleCI.
+
+## Set up CI for JHipster with CircleCI
+
+First, create the CircleCI configuration for the `store` microservice and the `gateway`.
+
+```shell
+cd store
+jhipster ci-cd
+```
+```shell
+cd gateway
+jhipster ci-cd
+```
+
+For both applications, choose the following options:
+- What CI/CD pipeline do you want to generate? **CircleCI**
+- What tasks/integrations do you want to include? (none)
+
+You must tweak the generated configuration at `store/.circleci/config.yml`, as the following changes are required for successful execution:
+- Change the execution environment of the workflow, to a dedicated VM, as that is required by the [TestContainers](https://www.testcontainers.org/supported_docker_environment/continuous_integration/circle_ci) dependency in tests.
+- As the dedicated VM includes docker, the docker installation step in the configuration must be removed
+- Add a step for building the docker container image
+- Add a step for pushing the image to DockerHub
+
+The final `config.yml` must look like this:
+
+```yml
+version: 2.1
+jobs:
+  build:
+    environment:
+      IMAGE_NAME: your-dockerhub-username/store
+    machine:
+      image: ubuntu-2004:current
+    resource_class: large
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            {%raw%}- v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+            # Perform a Partial Cache Restore (https://circleci.com/docs/2.0/caching/#restoring-cache)
+            - v1-dependencies-
+      - run:
+          name: Print Java Version
+          command: 'java -version'
+      - run:
+          name: Print Node Version
+          command: 'node -v'
+      - run:
+          name: Print NPM Version
+          command: 'npm -v'
+      - run:
+          name: Install Node Modules
+          command: 'npm install'
+      - save_cache:
+          paths:
+            - node
+            - node_modules
+            - ~/.gradle
+            key: {%raw%}v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+      - run:
+          name: Give Executable Power
+          command: 'chmod +x gradlew'
+      - run:
+          name: Backend tests
+          command: npm run ci:backend:test
```

Comment:
I tried adding `/home/circleci/.cache/Cypress` to the list of cache paths, but it still fails. I googled but was unable to figure out how to expire the cache.

Reply:
I had an issue too with e2e, but can't remember if it was the same. If you change the key prefix - v1 to - v2 a new cache will be created. I had to do that after changing to vm execution environment.

Reply:
I added the config.yml example for the gateway to the post, including front end tests.

I fixed the Cypress cache issue, but e2e tests fail. Both blog and store give 404 for all calls, all entity tests failing. It seems the test is not running the microservices. The same happens in my local. I believe the circleci generator leaves e2e disabled because of that. Also, the ci:e2e:teardown does not work in my local either (error while removing network).  The cypress cache is only required for e2e.